src/events.hpp

```diff
@@ -26,6 +26,7 @@
 #define DRAW_EVENT (SDL_USEREVENT + 3)
 #define CLOSE_WINDOW_EVENT (SDL_USEREVENT + 4)
 #define SHOW_HELPTIP_EVENT (SDL_USEREVENT + 5)
+#define TERMINATE_EVENT (SDL_USEREVENT + 6)
```

Comment:
The other alternative I thought of would be to throw the exception from the signal handler, but I'm not sure if it's safe to do that.